File: core/src/com/alexpi/awesometanks/entities/blocks/Wall.java
package com.alexpi.awesometanks.entities.blocks;

import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.physics.box2d.PolygonShape;
import com.badlogic.gdx.physics.box2d.World;

/**
 * Created by Alex on 13/01/2016.
 */
public class Wall extends Block {

    public Wall(AssetManager manager, World world, Vector2 pos){
        super(manager, "sprites/wall.png", world,new PolygonShape(),1,pos,1f, false, null);
    }


    //Makes it indestructible regardless of its health as long as its positive
    @Override
    public void takeDamage(float damage) { }
}

File: core/src/com/alexpi/awesometanks/entities/blocks/Wall.kt
package com.alexpi.awesometanks.entities.blocks

import com.alexpi.awesometanks.entities.components.body.BodyShape
import com.alexpi.awesometanks.entities.components.body.FixtureFilter
import com.alexpi.awesometanks.screens.game.stage.GameContext
import com.badlogic.gdx.math.Vector2

/**
 * Created by Alex on 13/01/2016.
 */
class Wall(gameContext: GameContext, pos: Vector2) : Block(gameContext,"sprites/wall.png", BodyShape.Box(1f, 1f), pos, FixtureFilter.BLOCK)